sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java

```diff
@@ -0,0 +1,282 @@
+                    // and set identities property in the payload model to null so that it won't be sent
+                    this.storageAccount.updateParameters.identity().withUserAssignedIdentities(null);
+                    return true;
+                } else {
+                    // Check user is asking to remove identities though there is no identities currently associated
+                    if (currentIds.isEmpty() && !removeIds.isEmpty() && currentIdentity == null) {
+                        // If so we are in a invalid state but we want to send user input to service and let service
+                        // handle it (ignore or error).
+                        currentIdentity = new Identity().withType(IdentityType.NONE).withUserAssignedIdentities(null);
+                        this.storageAccount.innerModel().withIdentity(currentIdentity);
+                        this.storageAccount.updateParameters.withIdentity(currentIdentity);
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Initialize storage account's identity property.
+     *
+     * @param identityType the identity type to set
+     */
+    private void initStorageAccountIdentity(IdentityType identityType) {
+        if (!identityType.equals(IdentityType.USER_ASSIGNED)
+            && !identityType.equals(IdentityType.SYSTEM_ASSIGNED)) {
+            throw logger.logExceptionAsError(new IllegalArgumentException("Invalid argument: " + identityType));
+        }
+        Identity currentIdentity = this.storageAccount.innerModel().identity();
+        if (this.storageAccount.isInCreateMode()) {
+            if (currentIdentity == null
+                || currentIdentity.type() == null
+                || currentIdentity.type().equals(IdentityType.NONE)
+                || currentIdentity.type().equals(identityType)) {
+                currentIdentity = new Identity().withType(identityType);
+            } else {
+                currentIdentity.withType(IdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED);
+            }
+            this.storageAccount.innerModel().withIdentity(currentIdentity);
+            this.storageAccount.createParameters.withIdentity(currentIdentity);
+        } else {
+            if (currentIdentity == null
+                || currentIdentity.type() == null
+                || currentIdentity.type().equals(IdentityType.NONE)
+                || currentIdentity.type().equals(identityType)) {
+                currentIdentity = Objects.isNull(currentIdentity) ? new Identity().withType(identityType) : currentIdentity.withType(identityType);
+            } else {
+                currentIdentity.withType(IdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED);
+            }
+            this.storageAccount.innerModel().withIdentity(currentIdentity);
+            this.storageAccount.updateParameters.withIdentity(currentIdentity);
```

Comment:
Does this work?

```java
            if (storageAccount.updateParameters.identity() == null) {
                storageAccount.updateParameters.withIdentity(storageAccount.innerModel().identity());
            }
            if (storageAccount.updateParameters.identity() == null) {
                storageAccount.updateParameters.withIdentity(new Identity());
            }

            if (storageAccount.updateParameters.identity().type() == null
                || storageAccount.updateParameters.identity().type().equals(IdentityType.NONE)
                || storageAccount.updateParameters.identity().type().equals(identityType)) {
                storageAccount.updateParameters.identity().withType(identityType);
            } else {
                storageAccount.updateParameters.identity().withType(IdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED);
            }
```

Reply:
The code related to `storageAccount.innerModel().identity()` has been deleted in the new version.